.github/workflows/automation-scripts/update-changelogs.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const path = require('path');
+
+// Read all files in the directory
+function getFilePaths(directoryPath) {
+  let filePaths = [];
+  let files = fs.readdirSync(directoryPath);
+  files.forEach(file => {
+    let filePath = path.join(directoryPath, file);
+    filePaths.push(filePath);
+  });
+  return filePaths
+}
+
+function loadJsonFiles(filePath) {
+  // Load each JSON file
+  data = fs.readFileSync(filePath);
+  const jsonData = JSON.parse(data);
+  return jsonData;
+}
+
+// returns true if currentVersion is greater than incoming
+function compareVersions(currentVersion, incomingVersion) {
+  currentVersion = currentVersion.split('.').map(Number);
+  incomingVersion = incomingVersion.split('.').map(Number);
+  const currentDif = currentVersion.map((num, index) => {
+    return num - incomingVersion[index];
+  })
+  const leadNum = currentDif.filter((num) => { return num !== 0 })[0];
+  return leadNum > 0;
+}
+
+function findNewEntries(currentEntries, incomingEntries) {
+  currVersions = [];
+  incomingVersions = [];
+
+  currentEntries.forEach((entry) => {
+    currVersions.push(entry.version);
+  });
+
+  incomingEntries.forEach((entry) => {
+    incomingVersions.push(entry.version);
+  });
+
+  newEntries = [];
```

Comment:
this is supposed to be just a set difference right?

Reply:
yes this is all the that exist in the incoming branch, but do not exist in the target branch. Note that in the case of a backported version like release/3.7 this branch is missing all the entries from 4.x.x, so this is why it is only looking for elements that are not in target.